src/coreclr/jit/emit.h

```diff
@@ -971,19 +971,25 @@ class emitter
                 iiaEncodedInstrCount = (count << iaut_SHIFT) | iaut_INST_COUNT;
             }
 
-#ifdef TARGET_ARMARCH
-
+#ifdef TARGET_ARM
             struct
             {
-#ifdef TARGET_ARM64
-                // For 64-bit architecture this 32-bit structure can pack with these unsigned bit fields
+                regNumber _idReg3 : REGNUM_BITS;
+                regNumber _idReg4 : REGNUM_BITS;
+            };
+#elif defined(TARGET_ARM64)
+            struct
+            {
+                // This 32-bit structure can pack with these unsigned bit fields
                 emitLclVarAddr iiaLclVar;
                 unsigned       _idRegBit : 1; // Reg3 is scaled by idOpSize bits
                 GCtype         _idGCref2 : 2;
-#endif
-                regNumber _idReg3 : REGNUM_BITS;
-                regNumber _idReg4 : REGNUM_BITS;
+                regNumber      _idReg3 : REGNUM_BITS;
+                regNumber      _idReg4 : REGNUM_BITS;
             };
+
+            insSvePattern _idSvePattern;
```

Comment:
this looks ok to me, but wondering why not just have something like this after the end of `struct` definition.

```c++
#ifdef TARGET_ARM64
  insSvePattern _idSvePattern;
#endif
```

Reply:
Happy to switch to that.

My reasoning was when you look at the whole file we have an `#ifdef` for each target. Many of them just have the same thing (`_idReg3` and `_idReg4`). The nesting of the Arm32/Arm64 is a little messy to read and doesn't fit the rest of the file style. Splitting it made it simpler.